Makefile

```diff
@@ -1,24 +1,36 @@
 
 DIST_DIR ?= $(CURDIR)/dist
+CHART_DIR ?= $(CURDIR)/traefik
 TMPDIR ?= /tmp
-SHIM_DIR ?= $(TMPDIR)/traefik
 HELM_REPO ?= $(CURDIR)/repo
-
+LINT_USE_DOCKER ?= true
+LINT_CMD ?= ct lint --config=ct.yaml
 ################################## Functionnal targets
 
 # Default Target: run all
 all: clean lint build deploy
 
-# Ensure the Helm chart and its metadata are valid
-lint: helm $(SHIM_DIR)
+# Ensure the Helm chart, YAMLs and metadatas are valid
+
+ifeq ($(LINT_USE_DOCKER),true)
+lint: docker
+else
+lint: helm ct
+endif
 	@echo "== Linting Chart..."
-	@helm lint $(SHIM_DIR)
+	@git remote add traefik https://github.com/containous/traefik-helm-chart >/dev/null 2>&1 || true
```

Comment:
Could you explain this line?

Reply:
Version increment check is done by `ct`, and we have to provide the name of a remote for this in `test/ct.yaml` (or to the `ct` CLI with a flag).

We cannot be sure that the original remote always have the same name for contributors (example: they don't use the pattern `upstream`, or fork of a fork, etc.).

This code ensure there is always a remote named `traefik` to compare to, which is the original one. Please note that it's expected to be the case on contributors repository, as we expect the version check to be run when they execute a `make lint` locally.

I took this trick from the `helm/charts` repo for information.

Maybe there is another way (or a better way) to do this?